### DISCUSSION TITLE
Permissions and graphql in swell.

### DISCUSSION THREAD

**Discussion by holden** [swellstores/community#358] [Help]
Hi,

I'm still trying to get a grasp on permissions in swell:

I added customer reviews onto my products model and everything is there inside the model except I decided to try and store the customer logos in their own model, `content/logos` so as not to store the same logo multiple times the problem is how can I access the data in this connected model?

eg I have:

```
query getProductBySlug {
  productBySlug(slug: "evenchilada-feud") {
    id
    name
    slug
    price
    content {
      brandId
      productReviews {
        name
        quote
        logoId
      }
    }
  }
}
```
![CleanShot 2023-08-28 at 10 50 33](https://github.com/swellstores/community/assets/48241/5dac0a29-c5c3-45a8-abba-0663c3755a7a)

This works fine, but for the logos can I use the lookup feature to now add these? The only method I currently am able to access is `logoId` is this possible or do I then need to do an additional query just on `/content/logos`?

![CleanShot 2023-08-28 at 10 59 42](https://github.com/swellstores/community/assets/48241/3011dd7a-e677-4d04-91ab-1b9a5366f144)

**Comment by holden** [comment]
When I try to query it directly it returns a count of 0

```
{
  contentLogos {
    count 
  }
}
```

**Reply by holden** [reply]
This works fine for `contentPages` but not for contentLogos which has 140+ records.

I also tried giving the frontend API key permissions

```
  "permissions": {
    "/content/logos": {
      "fields": ["name", "slug", "file"]
      }
    }
```

Nothing works... what am I missing here?

**Reply by logeshswell** [reply]
@sags95 can help you with it.

**Reply by holden** [reply]
That would be great. Thanks.

**Reply by holden** [reply]
There's obviously some kind of permissions issue or simple setting because even the graphql playground sees the models in the schema tab and autocompletes the correct names and things.

What do we need to do to be able to query these things using the frontend api keys the same way we do for other models?

```
{
  contentLogos {
    results {
      name
    }
  }
}
```

it just comes back null, when there are 142.

```
{
  "data": {
    "contentLogos": {
      "results": []
    }
  }
}
```

**Reply by sags95** [reply]
@holden I'm investigating an issue around content models with gql endpoint, I'll be sure to update this thread with updates as it moves through our process.

**Reply by holden** [reply]
Great, thanks for the update.
